wdl-cli/src/inputs.rs

```diff
@@ -0,0 +1,507 @@
+    ///
+    /// * If the input is a [`Input::File`], a reference to the inner path is
+    ///   returned wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn as_file(&self) -> Option<&Path> {
+        match self {
+            Input::File(p) => Some(p.as_path()),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and attempts to return the inner [`PathBuf`].
+    ///
+    /// * If the input is a [`Input::File`], the inner path buffer is returned
+    ///   wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn into_file(self) -> Option<PathBuf> {
+        match self {
+            Input::File(p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and returns the inner [`PathBuf`].
+    ///
+    /// # Panics
+    ///
+    /// If the input is not a [`Input::File`].
+    pub fn unwrap_file(self) -> PathBuf {
+        match self {
+            Input::File(p) => p,
+            v => panic!("{v:?} is not an `Input::File`"),
+        }
+    }
+
+    /// Attempts to return a reference to the inner key-value pair.
+    ///
+    /// * If the input is a [`Input::Pair`], a reference to the inner key and
+    ///   value is returned wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn as_pair(&self) -> Option<(&str, &Value)> {
+        match self {
+            Input::Pair { key, value } => Some((key.as_str(), value)),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and attempts to return the inner key-value pair.
+    ///
+    /// * If the input is a [`Input::Pair`], the inner key-value pair is
+    ///   returned wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn into_pair(self) -> Option<(String, Value)> {
+        match self {
+            Input::Pair { key, value } => Some((key, value)),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and returns the inner key-value pair.
+    ///
+    /// # Panics
+    ///
+    /// If the input is not a [`Input::Pair`].
+    pub fn unwrap_pair(self) -> (String, Value) {
+        match self {
+            Input::Pair { key, value } => (key, value),
+            v => panic!("{v:?} is not an `Input::Pair`"),
+        }
+    }
+}
+
+impl FromStr for Input {
+    type Err = Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Error> {
+        if !s.contains("=") {
+            let path = PathBuf::from(s);
+
+            if !path.exists() {
+                return Err(Error::FileNotFound(path));
+            }
+
+            return Ok(Input::File(path));
+        }
+
+        let parts = s.split("=").collect::<Vec<_>>();
+
+        if parts.len() != 2 {
+            return Err(Error::InvalidPair {
+                pair: s.to_string(),
+                reason: format!(
+                    "expected exactly one equal sign (`=`), found {}",
+                    parts.len() - 1,
+                ),
+            });
+        }
```

Comment:
What about `sprocket run ... weird_param='Bash_hack=foo'`? Where `weird_param` is of type `String`?